packages/flutter/lib/src/gestures/drag_details.dart

```diff
@@ -139,14 +141,19 @@ class DragUpdateDetails {
     this.delta = Offset.zero,
     this.primaryDelta,
     required this.globalPosition,
+    this.kind,
     Offset? localPosition,
+    this.offsetFromOrigin = Offset.zero,
```

Comment:
Is this necessary? In the past we let the caller to calculate this if they need.

tangential to this pr, I just realized there are states in  the drag gesture reconginzer, will the internal state messed up if it is sharing between multiple textfields?

Reply:
I've done this to avoid having to save `lastStartDetails` and passing that to the drag update method to calculate the origin of the drag.

`LongPressMoveUpdateDetails` has a similar API, and we also use it in text selection. https://github.com/flutter/flutter/pull/109573#discussion_r975775224

I think it's a nice to have, to avoid saving the`DragStartDetails`.